sdk/analytics/src/main/java/com/klaviyo/analytics/state/StateSideEffects.kt

```diff
@@ -110,7 +116,29 @@ internal class StateSideEffects(
     }
 
     private fun afterApiRequest(request: ApiRequest) = when {
-        request is PushTokenApiRequest && request.status == KlaviyoApiRequest.Status.Failed -> {
+        request.responseCode == HTTP_BAD_REQUEST -> {
+            request.responseBody?.let { responseBody ->
+                val error = KlaviyoErrorResponseDecoder.fromJson(JSONObject(responseBody))
+                error.errors.find { it.title == ErrorResponse.INVALID_INPUT_TITLE }?.let { inputError ->
+                    when (inputError.source?.pointer) {
+                        KlaviyoErrorSource.EMAIL_PATH -> {
+                            (Registry.get<State>() as? KlaviyoState)?.resetEmail()
+                            Registry.log.warning("Invalid email - resetting email state to null")
+                        }
+                        KlaviyoErrorSource.PHONE_NUMBER_PATH -> {
+                            (Registry.get<State>() as? KlaviyoState)?.resetPhoneNumber()
```

Comment:
This also feels a bit gross, but I don't want to add anything to the public State interface that would allow users to do this.